ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -323,9 +357,166 @@ func (ipfs *Connector) pinLsHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	respBytes, _ := json.Marshal(pinLs)
+	resBytes, _ := json.Marshal(pinLs)
 	w.WriteHeader(http.StatusOK)
-	w.Write(respBytes)
+	w.Write(resBytes)
+}
+
+func (ipfs *Connector) addHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
I thought of that too. It would be useful, but then, I don't want ipfs to pin because cluster needs to decide where to pin, and it could be a completely different node(s). I could let ipfs pin first and then unpin after it is repinned somewhere else too if this feature existed. In the end what I really need to know is which of the returned things are top-level hashes.

I could force that cluster allocates the pin to this node and some other, but then, there is no point in having metrics and stuff if content gets forcefully pinned where you send it regardless of the pinning strategy.